compiler/rustc_const_eval/src/interpret/memory.rs

```diff
@@ -1211,4 +1206,18 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             err_ub!(DanglingIntPointer(offset, CheckInAllocMsg::InboundsTest)).into()
         })
     }
+
+    /// Check if a `src` ptr overlaps with a `dest` ptr.
+    #[inline(always)]
+    pub fn check_ptr_overlap(
+        src_id: AllocId,
+        src_offset: Size,
+        dest_id: AllocId,
+        dest_offset: Size,
+        size: Size,
+    ) -> bool {
+        let overlaps = |a, b| a <= b && b < a + size;
```

Comment:
What happens if the addition here overflows?

Reply:
Thanks! I think we can use `b - a` to avoid overflows since `a <= b`